crates/astria-conductor/src/celestia/verify.rs

```diff
@@ -0,0 +1,342 @@
+    },
+};
+use celestia_client::{
+    CelestiaRollupBlob,
+    CelestiaSequencerBlob,
+};
+use moka::future::Cache;
+use sequencer_client::{
+    tendermint::block::{
+        signed_header::SignedHeader,
+        Height as SequencerHeight,
+    },
+    tendermint_rpc,
+    Client as _,
+    HttpClient as SequencerClient,
+};
+use telemetry::display::base64;
+use tokio_util::task::JoinMap;
+use tracing::{
+    info,
+    warn,
+};
+use tryhard::{
+    backoff_strategies::BackoffStrategy,
+    retry_fn,
+    RetryFutureConfig,
+    RetryPolicy,
+};
+
+use super::{
+    block_verifier,
+    convert::ConvertedBlobs,
+};
+use crate::utils::flatten;
+
+pub(super) struct VerifiedBlobs {
+    celestia_height: u64,
+    header_blobs: Vec<CelestiaSequencerBlob>,
+    rollup_blobs: Vec<CelestiaRollupBlob>,
+}
+
+impl VerifiedBlobs {
+    pub(super) fn len_header_blobs(&self) -> usize {
+        self.header_blobs.len()
+    }
+
+    pub(super) fn len_rollup_blobs(&self) -> usize {
+        self.rollup_blobs.len()
+    }
+
+    pub(super) fn into_parts(self) -> (u64, Vec<CelestiaSequencerBlob>, Vec<CelestiaRollupBlob>) {
+        (self.celestia_height, self.header_blobs, self.rollup_blobs)
+    }
+}
+
+/// Task key to track verification of multiple [`CelestiaSequencerBlob`]s.
+///
+/// The index is necessary because two keys might have clashing hashes and heights.
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
+struct VerificationTaskKey {
+    index: usize,
+    block_hash: [u8; 32],
+    sequencer_height: SequencerHeight,
+}
+
+/// Returns blobs that could be verified against Sequencer.
+pub(super) async fn verify_and_filter_decoded_blobs(
+    blob_verifier: Arc<BlobVerifier>,
+    converted_blobs: ConvertedBlobs,
+) -> VerifiedBlobs {
+    let (celestia_height, header_blobs, rollup_blobs) = converted_blobs.into_parts();
+
+    let mut verification_tasks = JoinMap::new();
+    let mut verified_header_blobs = Vec::with_capacity(header_blobs.len());
+
+    for (index, blob) in header_blobs.into_iter().enumerate() {
+        verification_tasks.spawn(
+            VerificationTaskKey {
+                index,
+                block_hash: blob.block_hash(),
+                sequencer_height: blob.height(),
+            },
+            blob_verifier.clone().verify_header_blob(blob),
+        );
+    }
+
+    while let Some((key, verification_result)) = verification_tasks.join_next().await {
+        match flatten(verification_result) {
+            Ok(verified_blob) => verified_header_blobs.push(verified_blob),
+            Err(error) => {
+                info!(
+                    block_hash = %base64(&key.block_hash),
+                    sequencer_height = %key.sequencer_height,
+                    %error,
+                    "verification of sequencer blob failed; dropping it"
+                );
+            }
+        }
+    }
+
+    VerifiedBlobs {
+        celestia_height,
+        header_blobs: verified_header_blobs,
+        rollup_blobs,
```

Comment:
these look like theyre not getting verified? they appear to be verified in the caller in mod.rs in the call to `reconstruct_blocks_from_verified_blobs` so maybe rename this to `verify_headers_and_filter_decoded_blobs`

Reply:
or update the function comment to say it just verifies header blobs

Reply:
I renamed it to just `verify_header_blobs` and adjusted the doc comment on the function.